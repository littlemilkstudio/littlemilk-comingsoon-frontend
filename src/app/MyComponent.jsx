import React from 'react'

class MyComponent extends React.Component {
  render () {
    return (
      <div>
        <h2>MyComponent</h2>
        <button id="my-button-one">button</button>
        <span>fart</span>
        <button id="my-button-two">button</button>
      </div>
    )
  }
}

export default MyComponent
