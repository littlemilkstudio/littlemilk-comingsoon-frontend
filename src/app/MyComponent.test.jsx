import React from 'react'
import { mount, shallow } from 'enzyme'

import MyComponent from './MyComponent'

describe('MyComponent', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<MyComponent debug />)
    expect(component).toMatchSnapshot()
  })

  it('should render correctly with no props', () => {
    const component = shallow(<MyComponent />)
    expect(component).toMatchSnapshot()
  })

  it('should render banner text correctly with given strings', () => {
    const strings = ['one', 'two']
    const component = shallow(<MyComponent list={strings} />)
    expect(component).toMatchSnapshot()
  })

  it('should be possible to activate button with Spacebar', () => {
    const component = mount(<MyComponent />)
    component
      .find('button#my-button-one')
      .simulate('keydown', { keyCode: 32 })

    expect(component).toMatchSnapshot()
    component.unmount()
  })

  it('should be possible to activate button 2 with Spacebar', () => {
    const component = mount(<MyComponent />)
    component
      .find('button#my-button-two')
      .simulate('keydown', { keyCode: 32 })

    expect(component).toMatchSnapshot()
    component.unmount()
  })
})
