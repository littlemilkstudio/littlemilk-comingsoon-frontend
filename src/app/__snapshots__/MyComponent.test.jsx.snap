// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should be possible to activate button 2 with Spacebar 1`] = `
<MyComponent>
  <div>
    <h2>
      MyComponent
    </h2>
    <button
      id="my-button-one"
    >
      button
    </button>
    <span>
      fart
    </span>
    <button
      id="my-button-two"
    >
      button
    </button>
  </div>
</MyComponent>
`;

exports[`MyComponent should be possible to activate button with Spacebar 1`] = `
<MyComponent>
  <div>
    <h2>
      MyComponent
    </h2>
    <button
      id="my-button-one"
    >
      button
    </button>
    <span>
      fart
    </span>
    <button
      id="my-button-two"
    >
      button
    </button>
  </div>
</MyComponent>
`;

exports[`MyComponent should render banner text correctly with given strings 1`] = `
<div>
  <h2>
    MyComponent
  </h2>
  <button
    id="my-button-one"
  >
    button
  </button>
  <span>
    fart
  </span>
  <button
    id="my-button-two"
  >
    button
  </button>
</div>
`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `
<div>
  <h2>
    MyComponent
  </h2>
  <button
    id="my-button-one"
  >
    button
  </button>
  <span>
    fart
  </span>
  <button
    id="my-button-two"
  >
    button
  </button>
</div>
`;

exports[`MyComponent should render correctly with no props 1`] = `
<div>
  <h2>
    MyComponent
  </h2>
  <button
    id="my-button-one"
  >
    button
  </button>
  <span>
    fart
  </span>
  <button
    id="my-button-two"
  >
    button
  </button>
</div>
`;
